src/langcheck/augment/en/_gender/_gender.py

```diff
@@ -0,0 +1,122 @@
+from __future__ import annotations
+
+from collections.abc import Iterable
+from random import choice
+from types import MappingProxyType
+
+import nltk
+from nltk import pos_tag, word_tokenize
+from nltk.tokenize.treebank import TreebankWordDetokenizer
+
+from langcheck.augment.en._gender._gender_pronouns import (_PRONOUNS_DICT,
+                                                           _BaseGenderPronouns)
+
+try:
+    nltk.data.find('averaged_perceptron_tagger')
+except LookupError:
+    nltk.download('averaged_perceptron_tagger')
+
+# This dictionary is used to determine the form of the pronoun.
+# Note that his and hers are not included in this dictionary because they can be
+# either of two different forms depending on the context.
+_PRONOUNS_FORM_DICT = MappingProxyType({
+    "she": "subject",
+    "he": "subject",
+    "they": "subject",
+    "their": "dependent_possessive",
+    "him": "object",
+    "them": "object",
+    "hers": "independent_possessive",
+    "theirs": "independent_possessive",
+    "himself": "reflexive",
+    "herself": "reflexive",
+    "themselves": "reflexive"
+})
+
+
+def _get_pronoun_form(word: str, tag: str) -> str | None:
+    """Get pronoun form of the word."""
+    # Handle degenerated cases.
+    if tag == "PRP$":
+        # PRP$ tag denotes a possessive pronoun.
+        return "dependent_possessive"
+    if word.lower() == "her":
+        return "object"
+    if word.lower() == "his":
+        return "independent_possessive"
+    return _PRONOUNS_FORM_DICT.get(word.lower())
+
+
+def _replace_pronoun(word: str, tag: str, target_pronouns: _BaseGenderPronouns):
+    # When the word is not a pronoun, return the word itself.
+    if (pronoun_form := _get_pronoun_form(word, tag)) is None:
+        return word
+    # Replace the pronoun with the target pronoun with the same form.
+    replaced_pronoun = getattr(target_pronouns, pronoun_form)
```

Comment:
It might be just a matter of preference, but it would be easier to read for me if `target_pronouns` is a dict and you just get a value by the key-value access. Is there any benefit (e.g. related to types) by using `getattr` here?

Reply:
I just wanted to make _BaseGenderPronouns immutable, since it should not be change.
Since it is not allowed to inherit `MappingProxyType`, there are no convenient immutable and customizable dict subclass in standard library, as far as I know.

Reply:
Got it, let's keep using `getattr` then!!